homeassistant/components/tami4/coordinator.py

```diff
@@ -0,0 +1,61 @@
+"""Water quality coordinator for Tami4Edge."""
+from dataclasses import dataclass
+from datetime import date, timedelta
+import logging
+
+from Tami4EdgeAPI import Tami4EdgeAPI
+from Tami4EdgeAPI.water_quality import WaterQuality
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
```

Comment:
I suggest freezing it, as it is an error if someone tries to modify any class argument.

```suggestion
@dataclass(frozen=True)
```